server/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -360,4 +364,26 @@ public void test_partial_quoted_subscript() {
         symbol = executor.asSymbol("nested_obj.\"myObj['x']\"['AbC']");
         assertThat(symbol, isReference("myObj['x']['AbC']"));
     }
+
+    @Test
+    public void test_resolve_eq_function_for_text_types_with_length_limit_and_unbound() throws IOException {
+        var e = SQLExecutor.builder(clusterService)
+            .addTable("create table tbl (str varchar(3))")
+            .build();
+        var eq = (Function) e.asSymbol("tbl.str = 'abc'");
+        assertThat(
+            Lists2.map(eq.arguments(), Symbol::valueType),
+            contains(DataTypes.STRING, DataTypes.STRING));
+    }
+
+    @Test
+    public void test_resolve_eq_function_for_text_types_with_different_limit_length() throws IOException {
```

Comment:
Not sure if these additional tests here adds something here compared to the new signature binder and type tests. Also no strong opinion here..

Reply:
can drop it